metadata/src/main/java/org/apache/kafka/controller/QuorumController.java

```diff
@@ -986,6 +1012,13 @@ public void handleCommit(BatchReader<ApiMessageAndVersion> reader) {
                             batch.appendTimestamp(),
                             committedBytesSinceLastSnapshot + batch.sizeInBytes()
                         );
+
+                        if (offset >= raftClient.latestSnapshotId().map(OffsetAndEpoch::offset).orElse(0L)) {
+                            oldestCommittedLogOnlyAppendTimestamp = Math.min(
```

Comment:
Why "LogOnlyAppendTimestamp" instead of "LogAppendTimestamp"?

Reply:
It is the oldest timestamp in the log that is not included in a snapshot. I was thinking that `oldestCommittedLogAppendTimestamp` could mislead the reader if they don't read the description of that variable:
```java
      /**                                                                                                                                                                                                                                                 
       * Timestamp for the oldest record that was committed but not included in a snapshot.                                                                                                                                                               
       */                                                                                                                                                                                                                                                 
      private long oldestCommittedLogOnlyAppendTimestamp = Long.MAX_VALUE;
```

Reply:
nit: Maybe overkill, but how about something even more verbose like `oldestNonSnapshottedLogAppendTimestamp`.
The current name still had me read the description.

Reply:
I was thinking of a similar name when I was implementing this feature but I felt it was too verbose. I don't have a strong opinion here so I can change it to whatever the reader prefers.

Reply:
I do think `oldestNonSnapshottedLogAppendTimestamp` is a little clearer. Perhaps the fact that it is an append timestamp is already clear from context and we could use `oldestNonSnapshottedTimestamp`? I also don't feel strongly though.

Reply:
I picked `oldestNonSnapshottedTimestamp`. Updated the PR.l